cloudpickle/cloudpickle.py

```diff
@@ -799,14 +839,14 @@ def _gen_ellipsis():
 def _gen_not_implemented():
     return NotImplemented
 
-def _fill_function(func, globals, defaults, dict):
+def _fill_function(func, globals, defaults, dict, closure):
     """ Fills in the rest of function data into the skeleton function object
         that were created via _make_skel_func().
          """
     func.__globals__.update(globals)
     func.__defaults__ = defaults
     func.__dict__ = dict
-
+    fill_cells(func.__closure__, closure)
```

Comment:
That is correct, this is a nop function on pypy where the original closure is passed to the skeleton function.